lib/Mojolicious/Plugin/OpenAPI.pm

```diff
@@ -65,14 +65,32 @@ sub register {
   $self;
 }
 
-sub _add_default_response {
-  my ($self, $op_spec, $code) = @_;
-  return if $op_spec->{responses}{$code};
-  my $name   = $self->{default_response_name};
-  my $ref    = $self->validator->schema->data->{definitions}{$name} ||= $self->_default_schema;
-  my %schema = ('$ref' => "#/definitions/$name");
+sub _add_default_response_2 {
+  my ($self, $op_spec) = @_;
+  for my $code (@{$self->{default_response_codes}}) {
+    next if $op_spec->{responses}{$code};
+
+    my $name   = $self->{default_response_name};
+    my $ref    = $self->validator->schema->data->{definitions}{$name} ||= $self->_default_schema;
+    my %schema = ('$ref' => "#/definitions/$name");
+    tie %schema, 'JSON::Validator::Ref', $ref, $schema{'$ref'}, $schema{'$ref'};
+    $op_spec->{responses}{$code} = {description => 'Default response.', schema => \%schema};
+  }
+}
+
+sub _add_default_response_3 {
+  my ($self, $op_spec) = @_;
+  return unless @{$self->{default_response_codes}};
+
+  my $root_spec    = $self->validator->get(['/']);
+  my $default_name = $self->{default_response_name};
+  my $ref          = $root_spec->{components}{responses}{$default_name} ||= {
+    description => 'default Mojolicious::Plugin::OpenAPI response',
+    content     => {'application/json' => {schema => $self->_default_schema}},
+  };
+  my %schema = ('$ref' => "#/components/responses/$default_name");
   tie %schema, 'JSON::Validator::Ref', $ref, $schema{'$ref'}, $schema{'$ref'};
-  $op_spec->{responses}{$code} = {description => 'Default response.', schema => \%schema};
+  $op_spec->{responses}{default} ||= \%schema;
```

Comment:
I don't think this reflects what happens in _add_default_response_2(). The code must loop over the "default_response_codes" to make sure the module does not respond to 201 or something else by accident.

Reply:
I don't see how plugin can respond with 201 (for example) by accident. If you specify this http status code in schema, it will be used in priority over default code. 
If you answer with 201 code and do not mention such response in schema, it's an error by itself, regardless of what plugin is doing under the hood.

The only way it can break something is when schema has its own default response. But actually it's the same problem plugin must deal with in v2. Schema can define it's own response for 401 error, but plugin makes assumption that developer knows what she/he is doing and somehow response will be altered before sending to client. 

I decided to use **default** response instead of code responses because I wanted to reduce size of generated json schema.

Reply:
Fix it according to your remark.